Implemented BindableLayout EmptyView

jfversluis opened the pull request:
### Description of Change ###

This change adds a property to the `BindableLayout` that lets users define an `EmptyView`. Whenever there are no items in the `BindableLayout.ItemsSource` the `EmptyView` or `EmptyViewTemplate` is shown.

### Issues Resolved ### 
<!-- Please use the format "fixes #xxxx" for each issue this PR addresses -->

- fixes #6932

### API Changes ###

Added:
 - object BindableLayout.EmptyView { get; set; } //Bindable Property
 - DataTemplate BindableLayout.EmptyViewTemplate {get; set; }  //Bindable Property

### Platforms Affected ### 
<!-- Please list all platforms affected by these changes -->

- Core/XAML (all platforms)

### Behavioral/Visual Changes ###
<!-- Describe any changes that may change how a user's app behaves or appears when upgrading to this version of the codebase. -->

Nothing happens if the user doesn't start using the new properties. When they do and the `BindableLayout.ItemsSource` is empty, the empty view will show up accordingly with how it is configured through the two properties.

By using the `EmptyView` the user can specify either a simple string that will be shown in a `Label` or define their own `View` that will be shown.

The user could also choose for the `EmptyViewTemplate` to show a `DataTemplate`. The `EmptyViewTemplate.BindingContext` will inherit the `BindingContext` from the `BindableLayout`.

When both the `EmptyView` and `EmptyViewTemplate` are set, the `EmptyViewTemplate` always takes precedence,

### Before/After Screenshots ### 
<!-- If possible, take a screenshot of your test case before these changes were made and another screenshot after the changes were made to show possible visual changes. -->

TBD

### Testing Procedure ###
<!-- Please list the steps that should be taken to properly test these changes on each relevant platform. If you were unable to test these changes yourself on any or all platforms, please let us know. Also, if you are able to attach a video of your test run, you will be our personal hero. -->

TBD

### PR Checklist ###
<!-- To be completed by reviewers -->

- [x] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

rmarinho commented on Xamarin.Forms.Core/BindableLayout.cs:
```diff
@@ -173,12 +230,28 @@ void CreateChildren()
+			if (_currentEmptyView == null)
+				return;
+
+			if (!_itemsSource.GetEnumerator().MoveNext())
```
_itemsSource can be null no?

jfversluis replied:
Good catch, thanks!

jfversluis commented:
@jsuarezruiz that looks a bit nicer! Done and a rebase just as a bonus.

rmarinho commented:
On iOS when i remove i still see the items and  it shows the empty view too.

![image](https://user-images.githubusercontent.com/1235097/68322870-21414a00-00bc-11ea-9cd4-28878727ef46.png)

![image](https://user-images.githubusercontent.com/1235097/68322889-30c09300-00bc-11ea-82d7-d63c7fa22cc1.png)

jfversluis commented:
Hm that is strange... Must be something that is changed then, this now also happens in the original BindableLayout gallery page. Let me have a look. Thanks!

jfversluis commented:
I have rebased and removed the shadow from this gallery page. The above bug is not because of this change.

I have created a separate issue for the "frame shadow"-bug here #8567, that way this PR can move forward
